src/status_im/chat/events/sign_up.cljs

```diff
@@ -49,37 +49,85 @@
                            :message-status :seen}}))
 
 (handlers/register-handler-fx
-  ::sign-up-success
-  [re-frame/trim-v (re-frame/inject-cofx :random-id)]
-  (fn [{:keys [db random-id]} message-id]
-    (-> {:db db
-         :dispatch-n [;; create manual way for entering confirmation code
-                      (sign-up/enter-confirmation-code-event random-id)
-                      ;; create automatic way for receiving confirmation code
-                      sign-up/start-listening-confirmation-code-sms-event]}
-        (message-seen message-id))))
+ :start-listening-confirmation-code-sms
+ [re-frame/trim-v]
+ (fn [{:keys [db]} [sms-listener]]
+   {:db (if-not (:confirmation-code-sms-listener db)
+          (assoc db :confirmation-code-sms-listener sms-listener)
+          db)}))
 
 (handlers/register-handler-fx
-  :start-listening-confirmation-code-sms
-  [re-frame/trim-v]
-  (fn [{:keys [db]} [sms-listener]]
-    {:db (if-not (:confirmation-code-sms-listener db)
-           (assoc db :confirmation-code-sms-listener sms-listener)
-           db)}))
+ :stop-listening-confirmation-code-sms
+ [re-frame/trim-v]
+ (fn [{:keys [db]}]
+   {:db                   (dissoc db :confirmation-code-sms-listener)
+    ::remove-sms-listener (:confirmation-code-sms-listener db)}))
+
+(re-frame/reg-fx
+ ::remove-sms-listener
+ (fn [subscription]
+   (sms-listener/remove-sms-listener subscription)))
+
+(defn- sms-receive-handler [{confirmation-code :body}]
+  (when-let [matches (re-matches #"(\d{4})" confirmation-code)]
+    (re-frame/dispatch [::sign-up-confirm (second matches)])))
+
+(def start-listening-confirmation-code-sms-event
+  [:request-permissions
+   [:receive-sms]
+   (fn []
+     (let [listener (sms-listener/add-sms-listener sms-receive-handler)]
+       (re-frame/dispatch [:start-listening-confirmation-code-sms listener])))])
+
+(handlers/register-handler-fx
+ ::sign-up-success
+ [re-frame/trim-v (re-frame/inject-cofx :random-id)]
+ (fn [{:keys [db random-id]} message-id]
+   (-> {:db         db
+        :dispatch-n [;; create manual way for entering confirmation code
+                     (sign-up/enter-confirmation-code-event random-id)
+                     ;; create automatic way for receiving confirmation code
+                     start-listening-confirmation-code-sms-event]}
+       (message-seen message-id))))
+
+(handlers/register-handler-fx
+ :update-phone-number-in-profile
```

Comment:
Where is this event actually used ? It's a new event and I don't see any place where it's actually dispatched.

Reply:
Yes, it can be removed 'cause it's a leftover. Thanks.